flink-runtime/src/main/java/org/apache/flink/runtime/state/AsyncSnapshotCallable.java

```diff
@@ -97,6 +97,10 @@ protected void cancel() {
 		}
 	}
 
+	protected CloseableRegistry getSnapshotCloseableRegistry() {
+		return snapshotCloseableRegistry;
```

Comment:
could we just make `snapshotCloseableRegistry` protected and use it directly in `RocksIncrementalSnapshotStrategy`? I think we do not need `(un)registerCloseableForCancellation` methods either afterwards and replace calls to them with direct calls to `snapshotCloseableRegistry.(un)registerCloseable`.